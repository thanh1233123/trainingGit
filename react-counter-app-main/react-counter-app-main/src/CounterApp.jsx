import PropTypes from 'prop-types';
import { useState } from 'react';


export const CounterApp = ({ value }) => {
  const [ counter, setCounter ] = useState( value );

  const handleAdd = () => setCounter( counter + 1 );
  const handleSubstract = () => setCounter( counter - 1 );
  const handleReset = () => setCounter( value );
  
  return (
    <>
      <h1>CounterApp</h1>
      <h2> { counter } </h2>

      <button onClick={ handleAdd } data-testid="btn-add"> +1 </button>
      <button onClick={ handleSubstract } data-testid="btn-substract"> -1 </button>
      <button onClick={ handleReset } data-testid="btn-reset"> Reset </button>
    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}