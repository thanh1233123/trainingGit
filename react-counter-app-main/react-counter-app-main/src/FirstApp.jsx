import PropTypes from 'prop-types';

const newMessage = [1, 2, 3, 4, 5, 6, 7, 8, 9];

const getResult = (a, b) => {
  return a + b;
}

// Promise object => render error
// const getResultAsync = async (a, b) => {
//   return a + b;
// }

export const FirstApp = ({ title, subTitle, version }) => {
  // console.log(props);

  return (
    <>
      <h1 data-testid="test-title">{ title }</h1>
      <h2>{ subTitle }</h2>
      <p>{ `Version: ${version}` }</p>
      <p>{ newMessage }</p>
      <h2>{ subTitle }</h2>
      <p>{ getResult(7, 2) }</p>
      {/* <p>{ getResultAsync(7, 2) }</p> */}
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  version: PropTypes.number
}

FirstApp.defaultProps = {
  subTitle: 'There is no subtitle',
  version: 1.0
}