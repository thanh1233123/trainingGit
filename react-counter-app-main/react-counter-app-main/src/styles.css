html, body {
  background-color: #21232A;
  color: white;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
  padding: 1rem;
}

button {
  padding: 15px;
  font-size: 1.3rem;
  margin-right: 6px;
  cursor: pointer;
}