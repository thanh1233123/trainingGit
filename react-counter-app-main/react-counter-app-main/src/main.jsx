import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';
import { CounterApp } from './CounterApp';


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value={ 10 } />
  </React.StrictMode>
);