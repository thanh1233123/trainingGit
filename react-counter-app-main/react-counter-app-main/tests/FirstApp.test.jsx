import { describe, it, expect } from 'vitest';
import { screen, render} from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Tests in <FirstApp />', () => {
  const title = 'A test title';
  const subTitle = 'A test subtitle';
  
  it('should match with snapshot', () => {
    const { container } = render( <FirstApp title={ title } /> );
    expect( container ).toMatchSnapshot();
  });

  it('should contain title', () => {
    render( <FirstApp title={ title } /> );
    expect( screen.getByText(title) ).toBeTruthy();
  });

  it('should show title on h1', () => {
    render( <FirstApp title={ title } /> );
    expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );
  });

  it('should contain props subTitle', () => {
    render( 
      <FirstApp 
        title={ title } 
        subTitle={ subTitle }
      /> 
    );
    
    expect( screen.getAllByText(subTitle).length ).toBe( 2 );
  });
});
