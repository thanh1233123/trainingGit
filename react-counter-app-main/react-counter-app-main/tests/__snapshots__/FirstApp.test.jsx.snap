// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests in <FirstApp /> > should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    A test title
  </h1>
  <h2>
    There is no subtitle
  </h2>
  <p>
    Version: 1
  </p>
  <p>
    1
    2
    3
    4
    5
    6
    7
    8
    9
  </p>
  <h2>
    There is no subtitle
  </h2>
  <p>
    9
  </p>
</div>
`;
